lib/internal/child_process.js

```diff
@@ -516,6 +516,12 @@ function setupChannel(target, channel) {
     // a message.
     target._send({ cmd: 'NODE_HANDLE_ACK' }, null, true);
 
+    // It can happen the handle is not received because of some error on
+    // ancillary data reception such as MSG_CTRUNC. In this case don't try to
+    // perform any conversion.
+    if (!handle)
```

Comment:
Right, I said "master" when I should have said "parent process."

You are right that unchecked repeated resending is no good.  The cluster module could deal with it by sending the handle to another worker after a few turns but that isn't applicable to the child_process module.

I don't have a great solution in mind but as a strawman: try three times and close + log a warning after that?  If it happens frequently enough and bothers users enough they will file bug reports about it and we can try to come up with something better.

We could cheat when the ChildProcess instance is managed by the cluster module and do the send-to-another-worker thing.  It would be tight coupling but that doesn't bother me personally.  Your call.